### PR TITLE
enhance(macros/APIRef): show badges on all links

### LINKED ISSUE #9430: Deprecated: should be listed as deprecated in the sidebar
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/DOMError

### What specific section or headline is this issue about?

side navigation

### What information was incorrect, unhelpful, or incomplete?

When on related API pages, such as https://developer.mozilla.org/en-US/docs/Web/API/DOMException, the DOMError API appears in the side navigation as a related API. It should have a deprecated icon next to it in the side bar.

### What did you expect to see?

an deprecated icon next to the link

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/domerror`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/DOMError
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/domerror/index.md
* Last commit: https://github.com/mdn/content/commit/f45409ba2169ff05e433d21aa4ee0424079916b8
* Document last modified: 2023-02-20T07:35:10.000Z

</details>

### PR DESCRIPTION
**Pull request by caugner**
## Summary

Fixes https://github.com/mdn/yari/issues/9430.

### Problem

The APIRef macros was showing badges (e.g. deprecated) only on some links. 

### Solution

Add badges to the remaining links.

---

## Screenshots

| Before | After |
|--------|--------|
| image | image |
| image | image |

---

## How did you test this change?

Ran `yarn && yarn dev` and opened http://localhost:3000/en-US/docs/Web/API/DOMException and http://localhost:3000/en-US/docs/Web/API/DOMError locally.

### CODE REVIEW

File: kumascript/macros/APIRef.ejs [hunk 1]
```diff
@@ -166,7 +166,13 @@ function getPageTitle(page) {
 }
 
 async function smartLinkWithBadges(url, content) {
-  const aPage = await wiki.getPage(url);
+  let aPage = await wiki.getPage(url);
+  console.log(env.locale, url, aPage);
```
Comment (caugner) [inline kumascript/macros/APIRef.ejs:170]:
```suggestion
```

File: kumascript/macros/APIRef.ejs [hunk 2]
```diff
@@ -165,6 +165,18 @@ function getPageTitle(page) {
   return title;
 }
 
+async function smartLinkWithBadges(url, content) {
+  let aPage = await wiki.getPage(url);
+  if (!aPage.title && env.locale !== 'en-US') {
+    const originalUrl = url.replace(`/${env.locale}/`, "/en-US/");
+    aPage = await wiki.getPage(originalUrl);
+    console.log("en-US", originalUrl, aPage);
```
Comment (caugner) [inline kumascript/macros/APIRef.ejs:173]:
```suggestion
```

**Review by caugner — COMMENTED** [review]
Self-approval to land this eventually.

**Comment by hamishwillee** [comment]
Thanks very much @caugner !!

**Pull request merged**